pkg/kubelet/kubelet_pods.go

```diff
@@ -1626,7 +1626,6 @@ func getPhase(pod *v1.Pod, info []v1.ContainerStatus, podIsTerminal bool) v1.Pod
 }
 
 func (kl *Kubelet) determinePodResizeStatus(pod *v1.Pod, podStatus *v1.PodStatus) v1.PodResizeStatus {
-	var podResizeStatus v1.PodResizeStatus
 	specStatusDiffer := false
 	for _, c := range pod.Spec.Containers {
 		if cs, ok := podutil.GetContainerStatus(podStatus.ContainerStatuses, c.Name); ok {
```

Comment:
@LingyanYin @Jeffwan @LastNight1997 Why we compare spec with status.ContainerStatuses.Resource, instead of status.ContainerStatuses.AllocatedResources?

Reply:
status.ContainerStatuses.Resource is the real status the container have now, and it's the original logic, we didn't change it.